substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/PosixJavaNetSubstitutions.java

```diff
@@ -119,84 +419,1339 @@ protected int peekData(DatagramPacket p) throws IOException {
+            //  1770          return iaObj;
+            return iaObj;
+        }
+
+        //  1773      /*
+        //  1774       * Map the Java level socket option to the platform specific
+        //  1775       * level and option name.
+        //  1776       */
+        //  1777      if (NET_MapSocketOption(opt, &level, &optname)) {
+        if (JavaNetNetUtilMD.NET_MapSocketOption(opt, level_Pointer, optname_Pointer) != 0) {
+            //  1778          JNU_ThrowByName(env, JNU_JAVANETPKG "SocketException", "Invalid option");
+            //  1779          return NULL;
+            throw new SocketException("Invalid option");
+        }
+
+        //  1782      if (opt == java_net_SocketOptions_IP_MULTICAST_LOOP &&
+        //  1783          level == IPPROTO_IP) {
+        if (opt == SocketOptions.IP_MULTICAST_LOOP && level_Pointer.read() == NetinetIn.IPPROTO_IP()) {
+            //  1784          optlen = sizeof(optval.c);
+            optlen_Pointer.write(SizeOf.get(CCharPointer.class));
+            optval_Pointer = optval_c_Pointer;
+        } else {
+            //  1786          optlen = sizeof(optval.i);
+            optlen_Pointer.write(SizeOf.get(CIntPointer.class));
+            optval_Pointer = optval_i_Pointer;
+        }
+        //  1788
+        //  1789      if (NET_GetSockOpt(fd, level, optname, (void *)&optval, &optlen) < 0) {
+        if (JavaNetNetUtilMD.NET_GetSockOpt(fd, level_Pointer.read(), optname_Pointer.read(), optval_Pointer, optlen_Pointer) < 0) {
+            //  1790          NET_ThrowByNameWithLastError(env, JNU_JAVANETPKG "SocketException",
+            //  1791                           "Error getting socket option");
+            //  1792          return NULL;
+            throw new SocketException("Error getting socket option");
+        }
+
+        //  1795      switch (opt) {
+        //  1796          case java_net_SocketOptions_IP_MULTICAST_LOOP:
+        if (opt == SocketOptions.IP_MULTICAST_LOOP) {
+            //  1798              if (level == IPPROTO_IP) {
+            if (level_Pointer.read() == NetinetIn.IPPROTO_IP()) {
+                //  1799                  return createBoolean(env, (int)!optval.c);
+                if (optval_c_Pointer.read() == 0) {
+                    return Boolean.TRUE;
+                }
+                return Boolean.FALSE;
+            } else {
+                //  1801                  return createBoolean(env, !optval.i);
+                if (optval_i_Pointer.read() == 0) {
+                    return Boolean.TRUE;
+                }
+                return Boolean.FALSE;
+            }
+        }
+        //  1804          case java_net_SocketOptions_SO_BROADCAST:
+        //  1805          case java_net_SocketOptions_SO_REUSEADDR:
+        else if (opt == SocketOptions.SO_BROADCAST || opt == SocketOptions.SO_REUSEADDR) {
+            //  1806              return createBoolean(env, optval.i);
+            if (optval_i_Pointer.read() == 0) {
```

Comment:
This is `optval.i`, not `!optval.i`.  See comment above about arguments to even (especially?) trivial methods.